changelog/16872.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
```

Comment:
Since the changes eliminate a redundant check and fixes tests hiding possible problems, but do not fix a logic error in the business code itself, may I suggest updating this to `improvement` instead of `bug`?

Reply:
Sure, changed to `improvement` 👍